### PR TITLE
changed tutorial so that it works

### PR DESCRIPTION
**Pull request by fiarabbit**
changed tutorial to cope with #3136

### CODE REVIEW

**Comment by fiarabbit** [comment]
This code is copy & paste from chainer/examples/ptb/train_ptb.py

**Comment by fiarabbit** [comment]
@niboshi It seems travis failed due to a function which I'd never modified. Could you please run travis again? The following is the error message.

```
======================================================================
FAIL: test_expm1_backward_cpu (test_exponential_m1.transplant_class.<locals>.C)  parameter: {'shape': (3, 2)}
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.5.1/lib/python3.5/site-packages/chainer/testing/condition.py", line 74, in wrapper
    fail()
  File "/home/travis/virtualenv/python3.5.1/lib/python3.5/site-packages/chainer/testing/condition.py", line 51, in fail
    instance.fail(msg)
AssertionError: 
Fail: 3, Success: 0
The first error message:
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.5.1/lib/python3.5/site-packages/chainer/testing/condition.py", line 60, in <lambda>
    lambda: f(ins, *args[1:], **kwargs),
  File "/home/travis/build/chainer/chainer/tests/chainer_tests/functions_tests/math_tests/test_exponential_m1.py", line 43, in test_expm1_backward_cpu
    self.check_backward(self.x, self.gy)
  File "/home/travis/build/chainer/chainer/tests/chainer_tests/functions_tests/math_tests/test_exponential_m1.py", line 39, in check_backward
    gradient_check.check_backward(F.expm1, x_data, y_grad)
  File "/home/travis/virtualenv/python3.5.1/lib/python3.5/site-packages/chainer/gradient_check.py", line 339, in check_backward
    testing.assert_allclose(gx, gx_accum, atol=atol, rtol=rtol)
  File "/home/travis/virtualenv/python3.5.1/lib/python3.5/site-packages/chainer/testing/array.py", line 24, in assert_allclose
    x, y, atol=atol, rtol=rtol, verbose=verbose)
  File "/home/travis/virtualenv/python3.5.1/lib/python3.5/site-packages/numpy/testing/utils.py", line 1395, in assert_allclose
    verbose=verbose, header=header)
  File "/home/travis/virtualenv/python3.5.1/lib/python3.5/site-packages/numpy/testing/utils.py", line 737, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=0.0001, atol=1e-05
(mismatch 100.0%)
 x: array(0.6710577290505171)
 y: array(0.670915424823761)
-------------------- >> begin captured stdout << ---------------------
error: 0.000142304226756
error: 0.000114284455776
error: 5.28628006577e-05
--------------------- >> end captured stdout << ----------------------
```

**Comment by niboshi** [comment]
@fiarabbit 
This error is caused by non-deterministic tests, and not relevant to the PR.
We're going to fix it soon, but please ignore it for now. I'm sorry for incovenience.

**Comment by fiarabbit** [comment]
@niboshi OK, thanks!

**Comment by fiarabbit** [comment]
@niboshi Would you please review this PR

**Comment by niboshi** [comment]
@fiarabbit  I'm sorry for delay. I'll review.

File: docs/source/tutorial/recurrentnet.rst [hunk 1]
```diff
@@ -291,21 +291,40 @@ Backprop Through Time is implemented as follows.
...
-       loss.backward()
-       loss.unchain_backward()  # truncate
-       optimizer.update()
-
-   updater = training.StandardUpdater(train_iter, optimizer, update_bptt)
-
-In this case, we update the parameters on every 35 consecutive words.
+   class BPTTUpdater(training.StandardUpdater):
+
+       def __init__(self, train_iter, optimizer, bprop_len, device):
```
Comment (niboshi) [inline docs/source/tutorial/recurrentnet.rst]: Could you remove `device` argument? This is not essential for this tutorial and may confuse users.

File: docs/source/tutorial/recurrentnet.rst [hunk 2]
```diff
@@ -291,21 +291,40 @@ Backprop Through Time is implemented as follows.
...
+           super(BPTTUpdater, self).__init__(
+               train_iter, optimizer, device=device)
```
Comment (niboshi) [inline docs/source/tutorial/recurrentnet.rst]: Ditto. (remove `device`)

File: docs/source/tutorial/recurrentnet.rst [hunk 3]
```diff
@@ -291,21 +291,40 @@ Backprop Through Time is implemented as follows.
...
+
+           # Progress the dataset iterator for bprop_len words at each iteration.
+           for i in range(self.bprop_len):
+               # Get the next batch (a list of tuples of two word IDs)
+               batch = train_iter.__next__()
+
+               # Concatenate the word IDs to matrices and send them to the device
+               # self.converter does this job
+               # (it is chainer.dataset.concat_examples by default)
+               x, t = self.converter(batch, self.device)
```
Comment (niboshi) [inline docs/source/tutorial/recurrentnet.rst]: Ditto. (remove `device`)

File: docs/source/tutorial/recurrentnet.rst [hunk 4]
```diff
@@ -291,21 +291,40 @@ Backprop Through Time is implemented as follows.
...
+
+               # Compute the loss at this time step and accumulate it
+               loss += optimizer.target(chainer.Variable(x), chainer.Variable(t))
+
+           optimizer.target.cleargrads()  # Clear the parameter gradients
+           loss.backward()  # Backprop
+           loss.unchain_backward()  # Truncate the graph
+           optimizer.update()  # Update the parameters
+
+In this case, we update the parameters on every ``bprop_len`` consecutive words.
```
Comment (niboshi) [inline docs/source/tutorial/recurrentnet.rst]: Could you add `updater = BPTTUpdater(train_iter, optimizer, bprop_len)` at the bottom of code, to demonstrate how to instantiate this class?

**Comment by niboshi** [comment]
jenkins, test this please.

**Comment by niboshi** [comment]
LGTM!
Thank you very much!

**Pull request merged**